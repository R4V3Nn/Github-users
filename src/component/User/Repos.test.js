import React from 'react';
import { mount } from 'enzyme';

import { UserRepos } from './Repos';

describe('Repos component', () => {
	const props = {
		userRepos: [{
			id: 11374954,
			node_id: 'MDEwOlJlcG9zaXRvcnkxMTM3NDk1NA==',
			name: 'ace',
			full_name: 'evanphx/ace',
			private: false,
			html_url: 'https://github.com/evanphx/ace',
			description: 'Ace (Ajax.org Cloud9 Editor)',
			fork: true,
			url: 'https://api.github.com/repos/evanphx/ace',
			created_at: '2013-07-12T18:12:26Z',
			updated_at: '2018-02-18T20:56:11Z',
			pushed_at: '2013-07-12T07:04:43Z',
			git_url: 'git://github.com/evanphx/ace.git',
			clone_url: 'https://github.com/evanphx/ace.git',
			svn_url: 'https://github.com/evanphx/ace',
			homepage: 'http://ace.ajax.org',
			language: 'JavaScript',
		}, {
			id: 53301806,
			node_id: 'MDEwOlJlcG9zaXRvcnkxMTM3NDk1NA==',
			name: 'ace',
			full_name: 'evanphx/ace',
			private: false,
			html_url: 'https://github.com/evanphx/ace',
			description: 'Ace (Ajax.org Cloud9 Editor)',
			fork: true,
			url: 'https://api.github.com/repos/evanphx/ace',
			created_at: '2013-07-12T18:12:26Z',
			updated_at: '2018-02-18T20:56:11Z',
			pushed_at: '2013-07-12T07:04:43Z',
			git_url: 'git://github.com/evanphx/ace.git',
			clone_url: 'https://github.com/evanphx/ace.git',
			svn_url: 'https://github.com/evanphx/ace',
			homepage: 'http://ace.ajax.org',
			language: 'JavaScript',
		}],
		reposUrl: 'https://api.github.com/repos',
		filteredRepos: [],
		selectedIds: ['11374954'],
		isDrawerShown: false,
		loadRepos: jest.fn(),
		onItemToggle: jest.fn(),
		selectAll: jest.fn(),
		reposFilter: jest.fn(),
		deleteRepo: jest.fn(),
	};

	it('should render Repos component', () => {
		const wrapper = mount(<UserRepos {...props} />);

		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
