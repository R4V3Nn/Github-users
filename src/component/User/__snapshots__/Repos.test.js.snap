// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos component should render Repos component 1`] = `
<UserRepos
  deleteRepo={[MockFunction]}
  filteredRepos={Array []}
  isDrawerShown={false}
  loadRepos={
    [MockFunction] {
      "calls": Array [
        Array [
          "https://api.github.com/repos",
          5,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onItemToggle={[MockFunction]}
  reposFilter={[MockFunction]}
  reposUrl="https://api.github.com/repos"
  selectAll={[MockFunction]}
  selectedIds={
    Array [
      "11374954",
    ]
  }
  userRepos={
    Array [
      Object {
        "clone_url": "https://github.com/evanphx/ace.git",
        "created_at": "2013-07-12T18:12:26Z",
        "description": "Ace (Ajax.org Cloud9 Editor)",
        "fork": true,
        "full_name": "evanphx/ace",
        "git_url": "git://github.com/evanphx/ace.git",
        "homepage": "http://ace.ajax.org",
        "html_url": "https://github.com/evanphx/ace",
        "id": 11374954,
        "language": "JavaScript",
        "name": "ace",
        "node_id": "MDEwOlJlcG9zaXRvcnkxMTM3NDk1NA==",
        "private": false,
        "pushed_at": "2013-07-12T07:04:43Z",
        "svn_url": "https://github.com/evanphx/ace",
        "updated_at": "2018-02-18T20:56:11Z",
        "url": "https://api.github.com/repos/evanphx/ace",
      },
      Object {
        "clone_url": "https://github.com/evanphx/ace.git",
        "created_at": "2013-07-12T18:12:26Z",
        "description": "Ace (Ajax.org Cloud9 Editor)",
        "fork": true,
        "full_name": "evanphx/ace",
        "git_url": "git://github.com/evanphx/ace.git",
        "homepage": "http://ace.ajax.org",
        "html_url": "https://github.com/evanphx/ace",
        "id": 53301806,
        "language": "JavaScript",
        "name": "ace",
        "node_id": "MDEwOlJlcG9zaXRvcnkxMTM3NDk1NA==",
        "private": false,
        "pushed_at": "2013-07-12T07:04:43Z",
        "svn_url": "https://github.com/evanphx/ace",
        "updated_at": "2018-02-18T20:56:11Z",
        "url": "https://api.github.com/repos/evanphx/ace",
      },
    ]
  }
/>
`;
